vncviewer/vncviewer.man

```diff
@@ -165,6 +165,20 @@ Path to certificate revocation list to use in conjunction with
 \fI~/.config/tigervnc/x509_crl.pem\fP.
 .
 .TP
+.B \-X509CERT \fIpath\fP
+Path to client certificate to use when authenticating client using any
+of the X509 security schemes (X509None, X509Vnc, etc.). Must be in PEM
```

Comment:
I don't think just adding it to the existing security types will be the best user experience. The client has no way of knowing if the server will ask for a client certificate with this approach. That means the client cannot use a different security type, in case it doesn't have a certificate.

The easiest approach is probably to add a new `X509Cert` type. It would fit in the theme of specifying how the user authenticates separately.

@berrange, you've done a lot of the VeNCrypt stuff. Any insight?

Reply:
> The client has no way of knowing if the server will ask for a client certificate with this approach. That means the client cannot use a different security type, in case it doesn't have a certificate.

That is correct, but at the same time I don't think it is a real world problem. 

If a server has gone to the trouble of enabling VeNCrypt with X509 certs, and decided to mandate client certs, IMHO, it is highly unlikely that the server will also offer other VNC auth options in parallel which don't require certs.  IOW, I don't think the idea that the client could choose a different security type is actually going to happen in practice.

Never say never, of course, but it is highly unlikely IMHO.